src/lib/OS/Vma.cc

```diff
@@ -1,99 +1,150 @@
 #include "simeng/OS/Vma.hh"
 
+#include <sys/stat.h>
+
+#include <cstddef>
+
 namespace simeng {
 namespace OS {
+HostFileMMap HostBackedFileMMaps::mapfd(int fd, size_t len, off_t offset) {
+  if (offset & (PAGE_SIZE - 1)) {
+    std::cerr << "[SimEng:HostBackedFileMMaps] Failed to create Host backed "
+                 "file mapping. Offset is not aligned "
+                 "to page size: "
+              << offset << std::endl;
+    std::exit(1);
+  }
+  struct stat* statbuf = (struct stat*)malloc(sizeof(struct stat));
+  int fstatResult = fstat(fd, statbuf);
+  if (fstatResult < 0) {
+    std::cerr << "[SimEng:HostBackedFileMMaps] fstat failed: Cannot create "
+                 "host backed file mmap for file "
+                 "descriptor - "
+              << fd << std::endl;
+    free(statbuf);
+    std::exit(1);
+  }
+  off_t fstatFileSize = statbuf->st_size;
+  free(statbuf);
+  if (offset + len > fstatFileSize) {
+    std::cerr << "[SimEng:HostBackedFileMMaps] Tried to create host backed "
+                 "file mmap with offset and size greater "
+                 "than file size."
+              << std::endl;
+    std::exit(1);
+  }
+  if (len <= 0) {
+    std::cerr << "[SimEng:HostBackedFileMMaps] Cannot create host backed file "
+                 "mmap with size 0 for file "
+                 "descriptor: "
+              << fd << std::endl;
+    std::exit(1);
```

Comment:
Based on the other comment above, shouldn't you call `free(statBuf)` before these exit calls?

Reply:
no because I free(statbuf) before these two warnings are checked for, and have put a free(statbuf) statement in the check above it incase it fails

Reply:
But based on this https://github.com/UoB-HPC/SimEng/pull/283#discussion_r1112938294 wasn't the free(statbuf) before these 2 checks moved to the end of the function?

Reply:
Clarified this offline, does not apply given that statbuf is no longer used later on in this function